### DISCUSSION TITLE
(Re)order Macros

### DISCUSSION THREAD

**Discussion by Rickmaurinus** [TabularEditor/TabularEditor3#1389] [Ideas]
When creating macros, they automatically receive an ID and are ordered in the sequence they were created. There's currently no easy way to re-order them. It's often convenient to organize them in a certain order. It would be great if we could simply drag and drop them in the place that we want. Or change the ID numbers

**Comment by KornAlexander** [comment]
totally agree. Once again just like with the other proposal serializing would be awesome.

- IDs can be changed in the macroactions.json file for sure but not UI I believe
- ordering according to Name, ID, sequence they were created in is possible

The only way to fix this programatically is currently is the python script from Andreas Nordgaard. Check my blog: https://actionablereporting.com/2024/10/30/my-power-bi-toolbox-80-tabular-editor-macros-to-automate-data-model-development/
